tests/integration/mixer/policy/ratelimiting_test.go

```diff
@@ -59,171 +61,218 @@ func TestRateLimiting_RedisQuotaRollingWindow(t *testing.T) {
-		// less than 10, don't fail this test).
-		want := math.Floor(want200s * 0.90)
-		// check successes
-		if got200s < want {
-			attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
-				util.Fqdn(destinationService, bookInfoNameSpaceStr)),
-				fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 200),
-				fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "destination")}
-			t.Logf("prometheus values for istio_requests_total for 200's:\n%s",
-				util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
-			t.Errorf("Bad metric value for successful requests (200s): got %f, want at least %f", got200s, want)
-		}
-		want200s = math.Ceil(want200s * 1.05)
-		if got200s > want200s {
-			attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
-				util.Fqdn(destinationService, bookInfoNameSpaceStr)),
-				fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 200),
-				fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "destination")}
-			t.Logf("prometheus values for istio_requests_total for 200's:\n%s",
-				util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
-			t.Errorf("Bad metric value for successful requests (200s): got %f, want at most %f", got200s,
-				want200s)
-		}
-	})
+	got429s, got200s, err := util.FetchRequestCount(t, prom, destinationService, "", bookInfoNameSpaceStr,
+		totalRequestExpected)
+	if err != nil {
+		return fmt.Errorf("error getting request count err:%v", err)
+	}
+	if got429s == 0 {
+		attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
+			util.Fqdn(destinationService, bookInfoNameSpaceStr)),
+			fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 429),
+			fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "source")}
+		t.Logf("prometheus values for istio_requests_total for 429's:\n%s",
+			util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
+		return fmt.Errorf("could not find 429s")
+	}
+	want429s = math.Floor(want429s * 0.70)
+	// Store new 200's and 429's value in prior vars and subtract prior's values from one
+	// obtained from prometheus.
+	temp := got429s
+	got429s -= prior429s
+	prior429s = temp
+	temp = got200s
+	got200s -= prior200s
+	prior200s = temp
+	t.Logf("Actual 429s: %f (%f rps)", got429s, got429s/actualDuration)
+	// check resource exhausted
+	if got429s < want429s {
+		attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
+			util.Fqdn(destinationService, bookInfoNameSpaceStr)),
+			fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 429),
+			fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "source")}
+		t.Logf("prometheus values for istio_requests_total for 429's:\n%s",
+			util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
+		return fmt.Errorf("bad metric value for rate-limited requests (429s): got %f, want at least %f", got429s,
+			want429s)
```

Comment:
this seems to be repeated in some form a bit. Can we abstract a bit and move to a func?